rust/src/domains/polars/datetime/ffi.rs

```diff
@@ -0,0 +1,53 @@
+use std::ffi::c_char;
+
+use opendp_derive::bootstrap;
+use polars::prelude::TimeUnit;
+
+use crate::{
+    core::FfiResult,
+    error::Fallible,
+    ffi::{any::AnyDomain, util},
+};
+
+use super::DatetimeDomain;
+
+#[bootstrap(
+    arguments(time_zone(default = b"null")),
+    generics(DI(suppress)),
+    returns(c_type = "FfiResult<AnyDomain *>")
+)]
+/// Construct an instance of `DatetimeDomain`.
+///
+/// Documentation on valid time zones can be found [in the Polars documentation](https://docs.pola.rs/user-guide/transformations/time-series/timezones/).
+///
+/// # Arguments
+/// * `time_unit` - One of `ns`, `us` or `ms`, corresponding to nano-, micro-, and milliseconds
```

Comment:
If it's possible with the macro, would we want to provide a default? (I would lean towards milliseconds.)